### PR TITLE
Extract fail_objective and pruned_objective for tests

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
There are duplicate objective definitions spread over tests. We extract `fail_objective`s and `pruned_objective`s to `testing/objective.py`.

## Description of the changes
* Extract `fail_objective`s and `pruned_objective`s to `testing/objective.py`.

### CODE REVIEW

File: optuna/testing/objectives.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from optuna import TrialPruned
+from optuna.trial import Trial
+
+
+def fail_objective(_: Trial) -> float:
+    raise ValueError
+
+
+def pruned_objective(trial: Trial) -> float:
+    raise TrialPruned()
```
Comment (toshihikoyanase) [inline optuna/testing/objectives.py:10]: `pruned_objective` is only used in `tests/test_callbacks.py`, and I don't think we need to extract it.

Comment (contramundum53) [reply optuna/testing/objectives.py:10]: Sorry, I forgot to replace four more places. PTAL.

**Review by not522 — COMMENTED** [review]
We can replace some functions in `study_tests/test_study.py` and `study_tests/test_optimize.py` with `fail_objective`.

File: optuna/testing/objectives.py (hunk 1 above)
Comment (toshihikoyanase) [reply optuna/testing/objectives.py]: Thanks!

File: optuna/testing/objectives.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+) -> Callable[[Trial], float]:
+    def func(trial: Trial) -> float:
+        for i, intermediate_value in enumerate(intermediate):
+            trial.report(step=i + 1, value=intermediate_value)
+        raise TrialPruned
+
+    return func
+
+
+pruned_objective = pruned_objective_with_intermediate_value([])
```
Comment (toshihikoyanase) [inline optuna/testing/objectives.py:23]: I feel the implementation of `pruned_objective` is a bit hacky. How about splitting the PR and focusing on `fail_objective` and `pruned_objective` as title mentioned?

Comment (contramundum53) [reply optuna/testing/objectives.py]: I reverted some of the last changes. PTAL.

**Review by not522 — APPROVED** [review]
LGTM

**Comment by toshihikoyanase** [comment]
The `checks` job failed. Could you check it, please?
```
Run flake8 .
./tests/study_tests/test_study.py:32:1: F401 'optuna.testing.objectives' imported but unused
1     F401 'optuna.testing.objectives' imported but unused
Error: Process completed with exit code 1.
```

**Comment by contramundum53** [comment]
Fixed.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM.

**Pull request merged**